addons/assigngear/functions/fnc_assignGear.sqf

```diff
@@ -90,58 +97,87 @@ _unit setUnitLoadout (configFile >> 'EmptyLoadout');
                     };
                 } forEach _x;
             };
-            case 11: { // magazines
+            case IDX_MAGAZINES: { // magazines
+                if (count _primaryMagazine > 0) then {
+                    if !(_primaryWeapon isEqualTo '') then {_unit addWeapon _primaryWeapon};
+                    { _unit addPrimaryWeaponItem _x; } forEach _primaryMagazine;
+                };
+                if (count _secondaryMagazine > 0) then {
+                    if !(_secondaryWeapon isEqualTo '') then {_unit addWeapon _secondaryWeapon};
+                    { _unit addSecondaryWeaponItem _x; } forEach _secondaryMagazine;
+                };
+                if (count _sidearmMagazine > 0) then {
+                    if !(_sidearmWeapon isEqualTo '') then {_unit addWeapon _sidearmWeapon};
+                    { _unit addHandgunItem _x; } forEach _sidearmMagazine;
+                };
+
                 { // Magazines try to fill vest first
                     switch true do {
                         case (_unit canAddItemToVest _x): {_unit addItemToVest _x;};
                         case (_unit canAddItemToUniform _x): {_unit addItemToUniform _x;};
                         default {_unit addItemToBackpack _x;};
                     };
                 } forEach _x;
+
+                if (count _primaryMagazine == 0) then {
+                    if !(_primaryWeapon isEqualTo '') then {_unit addWeapon _primaryWeapon};
+                };
+                if (count _secondaryWeapon == 0) then {
```

Comment:
`(count _secondaryWeapon == 0)` should be `(count _secondaryMagazine == 0)`

Reply:
😂

Reply:
This is what I get for coding after work. Cheers.